NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonPadding: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+            
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let previousButton = buttonsArray.filter{$0.tag == oldValue}.first
```

Comment:
Use `if let` block here, so you don't have to deal with optional `previousButton?` afterwards.